include/cosim/serialization.hpp

```diff
@@ -100,11 +100,16 @@ struct node_data_translator
 
 }} // namespace cosim::serialization
 
+/**
+ *  Serializes `data` into CBOR format and writes it to the output stream `out`.
+ */
+std::ostream& operator<<(std::ostream& out, const cosim::serialization::node& data);
```

Comment:
Something doesn't feel right about overloading `<<` and `>>` for this purpose. What if we want to add support for more serialisation formats? What if we want to add some additional settings to control how the serialisation is done?

Reply:
Update the code so that we can now do something like: 
```cpp
os << cosim::serialization::format::cbor << data
```
So that another format can be added if needed.